pkg/queryfrontend/cache/memcached.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package cache
```

Comment:
Do we need separate client? We kind of have the bucket cache with this already? :thinking:

Reply:
The bucket memcached cache already has `Store` and `Fetch` method, same name but different function signatures. So I cannot implement the Cortex interface on it.

Reply:
Hm, I am not sure about it - I think we can reuse them - let me check

Reply:
As talked offline, we can propose proper interface change (:

Reply:
@bwplotka @yeya24 I'd love to contribute to this interface as well. I had some struggles while integrating with groupcache so please include me into the discussions as well.

Reply:
Current interface

``` go
// Thanos
type Cache interface {
	Store(ctx context.Context, data map[string][]byte, ttl time.Duration)
	Fetch(ctx context.Context, keys []string) map[string][]byte
}

// Cortex
type Cache interface {
	Store(ctx context.Context, key []string, buf [][]byte)
	Fetch(ctx context.Context, keys []string) (found []string, bufs [][]byte, missing []string)
	Stop()
}
```

After checking some Cortex cache code, I think a better interface might be below. What do you think? IMHO `ttl` is specific to different cache, so it is better to not put it in the interface.

``` go
type Cache interface {
	Store(ctx context.Context, data map[string][]byte)
	Fetch(ctx context.Context, keys []string) (results map[string][]byte, missing []string)
	Stop()
}

```

Reply:
This looks good, cc @krasi-georgiev who might take it now